### PR TITLE
Compadre: Update the License and Copyright

### LINKED ISSUE #13173: Compadre:  Update the License and Copyright
## Documentation
Update the License and Copyright files to
* Reflect NTESS and new contract
* Utilize SPDX identifiers within files
* Make copyright and license files consistent across Trilinos packages (i.e., LICENSE and COPYRIGHT)

### PR DESCRIPTION
**Pull request by ccober6**
Update the License and Copyright files to

 * Reflect NTESS and new contract
 * Utilize SPDX identifiers within files
 * Make copyright and license files consistent across Trilinos packages (i.e., LICENSE and COPYRIGHT)

@trilinos/compadre 

* Closes #13173

### CODE REVIEW

File: packages/compadre/src/tpl/nanoflann.hpp [hunk 1]
```diff
@@ -1,34 +1,16 @@
...
- * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
- * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
- * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
- * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
- * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
- * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
- * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- *************************************************************************/
+// @HEADER
```
Comment (kuberry) [inline packages/compadre/src/tpl/nanoflann.hpp:1]: This is a BSD2 header for nanoflann (not an NTESS product)

**Comment by kuberry** [comment]
Hi @ccober6, I can't push commits to your repo, but could you apply the following patch? This will restore the nanoflann header and will rename COPYRIGHT as COPYRIGHT.md (to match LICENSE.md and README.md).

patch.txt
******************************
```sh
From e380a7e4219cb0bdf89372dcc90eccc19a8ff411 Mon Sep 17 00:00:00 2001
From: Paul Kuberry <pakuber@sandia.gov>
Date: Tue, 2 Jul 2024 12:10:34 -0400
Subject: [PATCH 1/1] compadre: restore nanoflann header and rename COPYRIGHT

---
 packages/compadre/{COPYRIGHT => COPYRIGHT.md} |  0
 packages/compadre/src/tpl/nanoflann.hpp       | 44 +++++++++++++------
 2 files changed, 31 insertions(+), 13 deletions(-)
 rename packages/compadre/{COPYRIGHT => COPYRIGHT.md} (100%)

diff --git a/packages/compadre/COPYRIGHT b/packages/compadre/COPYRIGHT.md
similarity index 100%
rename from packages/compadre/COPYRIGHT
rename to packages/compadre/COPYRIGHT.md
diff --git a/packages/compadre/src/tpl/nanoflann.hpp b/packages/compadre/src/tpl/nanoflann.hpp
index da731881d49..d169f95c9bb 100755
--- a/packages/compadre/src/tpl/nanoflann.hpp
+++ b/packages/compadre/src/tpl/nanoflann.hpp
@@ -1,16 +1,34 @@
-// @HEADER
-// *****************************************************************************
-//     Compadre: COMpatible PArticle Discretization and REmap Toolkit
-//
-// Copyright 2001       Nicolai M. Josuttis
-// Copyright 2008-2009  Marius Muja (mariusm@cs.ubc.ca). All rights reserved.
-// Copyright 2008-2009  David G. Lowe (lowe@cs.ubc.ca). All rights reserved.
-// Copyright 2011-2016  Jose Luis Blanco (joseluisblancoc@gmail.com).
-//                      All rights reserved.
-// Copyright 2018 NTESS and the Compadre contributors.
-// SPDX-License-Identifier: BSD-2-Clause
-// *****************************************************************************
-// @HEADER
+/***********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ * Copyright 2008-2009  Marius Muja (mariusm@cs.ubc.ca). All rights reserved.
+ * Copyright 2008-2009  David G. Lowe (lowe@cs.ubc.ca). All rights reserved.
+ * Copyright 2011-2016  Jose Luis Blanco (joseluisblancoc@gmail.com).
+ *   All rights reserved.
+ *
+ * THE BSD LICENSE
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *************************************************************************/
 
 /** \mainpage nanoflann C++ API documentation
   *  nanoflann is a C++ header-only library for building KD-Trees, mostly
-- 
2.39.3 (Apple Git-146)
```

**Comment by ccober6** [comment]
Hi @kuberry, I am ok with returning the original header, though I believe the new one minus the "// Copyright 2018 NTESS and the Compadre contributors." would also be correct.  I will use your patch and retest.

**Comment by ccober6** [comment]
@kuberry , thanks for catching this.  Can you re-review?  :)

**Review by kuberry — APPROVED** [review]

**Comment by kuberry** [comment]
> @kuberry , thanks for catching this. Can you re-review? :)

Sure thing. LGTM.

**Pull request merged**